packages/wallet/src/redux/selectors.ts

```diff
@@ -1,24 +1,18 @@
-import { OpenedState, OPENING, ChannelStatus } from './channel-state/state';
+import { OpenedState, OPENING, ChannelStatus, ChannelState } from './channel-state/state';
 import * as walletStates from './state';
 import * as indirectFundingStates from './indirect-funding/state';
 import { DirectFundingState } from './direct-funding/state';
 
-export const getOpenedChannelState = (
-  state: walletStates.Initialized,
-  channelId: string,
-): OpenedState => {
+export const getOpenedChannelState = (state: ChannelState, channelId: string): OpenedState => {
```

Comment:
I think the `state` variable should be as close to a wallet state as possible. It can, and probably should be of type `{sharedData: SharedData}`

Reply:
I'm a bit hesitant about this since it infers our `walletState` has the same shape as our `ProtocolWithSharedData`.

I prefer having the selectors more strongly typed against a specific type of state.

Reply:
Do you mean then that a selector that wants a specific channel state, given a wallet state, should call
`getOpenedChannelState(getChannelState(walletState), channelId)`? Or that they should call `getOpenedChannelState(walletState.channelState, channelId)`?

The second option goes against the philosophy that callers of a selector need to know something about the shape of the wallet state. This is ok if it's a "private" selector that's called in the context where you directly have a `channelState` object, but it shouldn't be used in a UI component, for instance.